lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/participants/codeaction/MavenCodeActionParticipant.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat Inc. and others.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.lemminx.extensions.maven.participants.codeaction;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.eclipse.lemminx.extensions.contentmodel.participants.XMLSyntaxErrorCode;
+import org.eclipse.lemminx.services.extensions.codeaction.ICodeActionParticipant;
+import org.eclipse.lemminx.services.extensions.codeaction.ICodeActionRequest;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+
+public class MavenCodeActionParticipant implements ICodeActionParticipant {
+	private final Map<String, ICodeActionParticipant> codeActionParticipants;
+
+	public MavenCodeActionParticipant() {
+		super();
+		codeActionParticipants = new HashMap<>();
+	}
+
+	@Override
+	public void doCodeAction(ICodeActionRequest request, List<CodeAction> codeActions, CancelChecker cancelChecker) {
+		Diagnostic diagnostic = request.getDiagnostic();
+		if (diagnostic == null || diagnostic.getCode() == null || !diagnostic.getCode().isLeft()) {
+			return;
+		}
```

Comment:
Code Actions are not only "Quick-fix" for errors, they can also be quick-edits. Diagnostic may be null yet some code-actions be possible (eg "extract version to variable")

Reply:
It was supposed to work only for diagnostics with codes... but again, I think it's a subject to change